x-pack/plugin/security/qa/multi-cluster/src/javaRestTest/java/org/elasticsearch/xpack/remotecluster/RemoteClusterSecurityRCS2PainlessExecuteIT.java

```diff
@@ -0,0 +1,321 @@
+    public static TestRule clusterRule = RuleChain.outerRule(new RunnableTestRuleAdapter(() -> {
+        SSL_ENABLED_REF.set(usually());
+        NODE1_RCS_SERVER_ENABLED.set(randomBoolean());
+        NODE2_RCS_SERVER_ENABLED.set(randomBoolean());
+        INVALID_SECRET_LENGTH.set(randomValueOtherThan(22, () -> randomIntBetween(0, 99)));
+    })).around(fulfillingCluster).around(queryCluster);
+
+    @SuppressWarnings({ "unchecked", "checkstyle:LineLength" })
+    public void testPainlessExecute() throws Exception {
+        configureRemoteCluster();
+
+        {
+            // Query cluster -> add role for test user - do not give any privileges for remote_indices
+            final var putRoleRequest = new Request("PUT", "/_security/role/" + REMOTE_SEARCH_ROLE);
+            putRoleRequest.setJsonEntity("""
+                {
+                  "indices": [
+                    {
+                      "names": ["local_index", "my_local*"],
+                      "privileges": ["read"]
+                    }
+                  ]
+                }""");
+            assertOK(adminClient().performRequest(putRoleRequest));
+
+            // Query cluster -> create user and assign role
+            final var putUserRequest = new Request("PUT", "/_security/user/" + REMOTE_SEARCH_USER);
+            putUserRequest.setJsonEntity("""
+                {
+                  "password": "x-pack-test-password",
+                  "roles" : ["remote_search"]
+                }""");
+            assertOK(adminClient().performRequest(putUserRequest));
+
+            // Query cluster -> create test index
+            final var indexDocRequest = new Request("POST", "/local_index/_doc?refresh=true");
+            indexDocRequest.setJsonEntity("{\"local_foo\": \"local_bar\"}");
+            assertOK(client().performRequest(indexDocRequest));
+
+            // Fulfilling cluster -> create test indices
+            final Request bulkRequest = new Request("POST", "/_bulk?refresh=true");
+            bulkRequest.setJsonEntity(Strings.format("""
+                { "index": { "_index": "index1" } }
+                { "foo": "bar" }
+                { "index": { "_index": "secretindex" } }
+                { "bar": "foo" }
+                """));
+            assertOK(performRequestAgainstFulfillingCluster(bulkRequest));
+        }
+
+        {
+            // TEST CASE 1: Query local cluster for local_index - should work since role has read perms for it
+            Request painlessExecuteLocal = createPainlessExecuteRequest("local_index");
+
+            Response response = performRequestWithRemoteSearchUser(painlessExecuteLocal);
+            assertOK(response);
+            String responseBody = EntityUtils.toString(response.getEntity());
+            assertThat(responseBody, equalTo("{\"result\":[\"test\"]}"));
+        }
+        {
+            // // TEST CASE 2: Query remote cluster for index1 - should fail since no permissions granted for remote clusters yet
```

Comment:
Let's decide what to do with these test cases before merging, rather than just leaving them commented out like this. If we care about them, let's get them working (or at least open an issue to report the problem they represent). If we don't care about them, let's just remove them.